ResearchKit/Common/ORKResultPredicate.h

```diff
@@ -250,6 +409,24 @@ ORK_CLASS_AVAILABLE
  
  @param taskIdentifier      The identifier of the task whose result you want to match. Pass `nil`
                                 to match the ongoing task.
+ @param stepIdentifier      The identifier of the step whose result you want to match. Passing `nil`
+                                assumes the step identifier is the same as the result identifier.
+ @param resultIdentifier    The identifier of the question result you are interested in.
+ @param pattern             An ICU-compliant regular expression pattern that matches the answer string.
+ 
+ @return A result predicate.
+ */
++ (NSPredicate *)predicateForChoiceQuestionResultWithTaskIdentifier:(nullable NSString *)taskIdentifier
+                                                     stepIdentifier:(nullable NSString *)stepIdentifier
+                                                   resultIdentifier:(NSString *)resultIdentifier
```

Comment:
Good idea!

The only downside is that the new methods won't be backwards compatible with the current `stable` branch, but I don't think that's too much of a problem. Is it?